capability/capability_linux.go

```diff
@@ -27,21 +28,32 @@ const (
 var (
 	capVers    uint32
 	capLastCap Cap
+	once       sync.Once
 )
 
-func init() {
-	var hdr capHeader
-	capget(&hdr, nil)
-	capVers = hdr.version
-
-	if initLastCap() == nil {
-		CAP_LAST_CAP = capLastCap
-		if capLastCap > 31 {
-			capUpperMask = (uint32(1) << (uint(capLastCap) - 31)) - 1
-		} else {
-			capUpperMask = 0
+func getCapMaskCapLast() (uint32, Cap) {
+	once.Do(func() {
+		var hdr capHeader
+		capget(&hdr, nil)
+		capVers = hdr.version
+
+		if initLastCap() == nil {
+			CAP_LAST_CAP = capLastCap
```

Comment:
I think you can remove CAP_LAST_CAP to be sure that old users notice the change and update their code.

BTW: @kolyshkin created the fork https://github.com/kolyshkin/capability and he is going to support it.